lib/fizzy/execute.cpp

```diff
@@ -340,12 +345,21 @@ inline bool load_from_memory(
     return true;
 }
 
+template <typename T>
+inline T pop_store_memory_arg(OperandStack& stack) noexcept
+{
+    if constexpr (std::is_floating_point_v<T>)
+        return stack.pop().as<T>();
+    else
+        return static_cast<T>(stack.pop().i64);
+}
+
 template <typename DstT>
 inline bool store_into_memory(
     bytes& memory, OperandStack& stack, const uint8_t*& immediates) noexcept
 {
-    const auto value = static_cast<DstT>(stack.pop());
-    const auto address = static_cast<uint32_t>(stack.pop());
+    const auto value = pop_store_memory_arg<DstT>(stack);
```

Comment:
Why `stack.pop().template as<DstT>()` does not work here?

Reply:
There's no `as` for `uint8_t`, `uint16_t`